sync/src/tests/relayer.rs

```diff
@@ -212,8 +217,16 @@ fn relay_compact_block_with_missing_indexs() {
                 .collect::<Vec<_>>();
 
             [3, 5].iter().for_each(|i| {
+                let tx = &txs[*i];
+                let cycles = {
+                    let chain_state = shared1.chain_state().lock();
```

Comment:
lock acquired in loop

Reply:
Its tests code, hold the lock outside will block the tests(the node is in the same process).